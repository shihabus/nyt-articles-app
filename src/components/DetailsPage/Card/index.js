import React from "react";
import { View, StyleSheet } from "react-native";
import PropTypes from "prop-types";
import Footer from './Footer'
import Content from './Content'
import Title from './Title'

const styles = StyleSheet.create({
  container: {
    width: "90%",
    alignSelf: "center",
    backgroundColor: "hsla(150, 15%, 71%, 0.5)",
    borderRadius: 10,
    top: -30,
    padding: 20
  }
});

export const DetailsCard = ({ params }) => {
  const { title, date, author, content, url, source } = params;
  return (
    <View style={styles.container}>
      <Title title={title} date={date} author={author} />
      <Content content={content} />
      <Footer url={url} source={source} />
    </View>
  );
};

DetailsCard.propTypes = {
  params: PropTypes.shape({
    title: PropTypes.string,
    date: PropTypes.string,
    author: PropTypes.string,
    content: PropTypes.string,
    url: PropTypes.string,
    source: PropTypes.string
  })
};
