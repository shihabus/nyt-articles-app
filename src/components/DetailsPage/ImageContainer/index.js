import React from "react";
import {  Image, StyleSheet } from "react-native";
import PropTypes from "prop-types";

const styles = StyleSheet.create({
  image: {
    height: 290,
    width: undefined,
    backgroundColor:'#d1d1d1'
  }
});

export const ImageSection = ({ url }) => {
  return (
    <Image
      style={styles.image}
      resizeMode="cover"
      source={{ uri: `${url}` }}
    />
  );
};

ImageSection.propTypes = {
  url: PropTypes.string
};

