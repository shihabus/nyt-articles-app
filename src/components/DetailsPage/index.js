export * from './ImageContainer'
export * from './Card'