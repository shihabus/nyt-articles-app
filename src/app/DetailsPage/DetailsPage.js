/* eslint-disable camelcase */
import React, { Component } from "react";
import { View } from "react-native";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import { ImageSection, DetailsCard } from "../../components/DetailsPage";
import { COLORS, STRINGS } from "../../constants";

class DetailsPage extends Component {
  static navigationOptions = () => {
    return {
      title: STRINGS.detailsPageTitle,
      headerStyle: {
        backgroundColor: COLORS.headerBar
      },
      headerTintColor: COLORS.headerTextColor,
      headerTitleStyle: {
        fontWeight: "normal",
        fontSize: 18
      }
    };
  };

  render() {
    const {
      selectedArticle: {
        abstract,
        byline,
        published_date,
        source,
        title,
        url,
        imageURL
      }
    } = this.props;

    const urlToImage = imageURL;

    const params = {
      title,
      date: published_date,
      author: byline,
      content: abstract,
      url,
      source
    };
    return (
      <View style={{ flex: 1 }}>
        <ImageSection url={urlToImage} testID='ImageSection'/>
        <DetailsCard params={params} testID='DetailsCard'/>
      </View>
    );
  }
}

DetailsPage.propTypes = {
  selectedArticle: PropTypes.shape({
    abstract: PropTypes.string,
    byline: PropTypes.string,
    published_date: PropTypes.string,
    source: PropTypes.string,
    title: PropTypes.string,
    url: PropTypes.string,
    imageURL: PropTypes.string
  })
};

const mapStateToProps = ({ selection }) => {
  const { selectedArticle } = selection;
  return { selectedArticle };
};

export default connect(
  mapStateToProps,
  null
)(DetailsPage);
