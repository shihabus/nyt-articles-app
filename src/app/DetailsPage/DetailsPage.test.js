/* eslint-disable no-unused-vars */
/* eslint-disable import/named */
/* eslint-disable no-undef */
/**
 * @jest-environment jsdom
 */

import React from "react";
import { createSerializer } from "enzyme-to-json";
import { connect } from "react-redux";
import DetailsPage, { mapStateToProps } from "./DetailsPage";
import { findComponent, testStore,checkProps } from "../../../testUtils";

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

const setUp = (initialStore = {}) => {
  const store = testStore(initialStore);
  const wrapper = shallow(<DetailsPage store={store} />)
    .childAt(0)
    .dive();
  return wrapper;
};

describe("Display Page", () => {
  let wrapper;
  beforeEach(() => {
    const initialStore = {
      selection: {
        selectedArticle: {
          abstract: "test",
          byline: "test",
          imageURL: "test",
          published_date: "2019-07-02",
          source: "The New York Times",
          title: "Test",
          url:
            "https://www.nytimes.com/2019/07/02/nyregion/judge-james-troiano-rape.html"
        }
      }
    };
    wrapper = setUp(initialStore);
  });

  it("Snapshot test", () => {
    expect(wrapper).toMatchSnapshot('Display page');
  });

  it('should render child components',()=>{
    expect(findComponent(wrapper,"ImageSection").length).toEqual(1)
    expect(findComponent(wrapper,"DetailsCard").length).toEqual(1)
  })

  it('Checked props',()=>{
    const expectedProps={
        selectedArticle: {
            abstract: 'Test',
            byline: 'Test',
            published_date: 'Test',
            source: 'Test',
            title: 'Test',
            url: 'Test',
            imageURL: 'Test'
          }
    }
    const propsErr=checkProps(DetailsPage,expectedProps)
    expect(propsErr).toBeUndefined()
  })
});
