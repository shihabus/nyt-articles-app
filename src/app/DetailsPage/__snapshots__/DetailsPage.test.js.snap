// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Page Snapshot test: Display page 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ImageSection
    testID="ImageSection"
    url="test"
  />
  <DetailsCard
    params={
      Object {
        "author": "test",
        "content": "test",
        "date": "2019-07-02",
        "source": "The New York Times",
        "title": "Test",
        "url": "https://www.nytimes.com/2019/07/02/nyregion/judge-james-troiano-rape.html",
      }
    }
    testID="DetailsCard"
  />
</View>
`;
